---
title: "ALFA-K results"
geometry: margin=0.5cm
output:
  pdf_document:
header-includes:
- \usepackage{caption}
- \captionsetup[figure]{labelformat=empty}
---

\section*{Abstract} 

Aneuploidy occurs in most solid tumors and has the potential to dramatically modify cellular phenotype and fitness. Despite the importance of aneuploidy in tumor evolution, quantitative understanding of the evolutionary landscape of aneuploidy is lacking. To address this question, we developed a method to infer the fitness landscape of chromosome level karyotypes. Our method takes longitudinal single cell karyotype data from an evolving cell population as input, then estimates the fitness of thousands of karyotypes located near to the input data in karyotype space. The predictive ability of the method was validated using artificial data generated from an agent based model, as well as data from a set of in vitro and in vivo passaged cell lines. Analysis of fitted landscapes illustrated how the fitness impact of a CNA depends on both cellular and biological context. Moreover missegregation rate can interact with fitness to determine clonal dominance. 

\section*{Introduction} 

Losses and gains of entire chromosomes or large sections thereof - termed aneuploidy - are a defining feature of solid tumors  \cite{Bakhoum2018-ai, Santaguida2015-bx}. Since an estimated 90\% of solid tumours are aneuploid and because aneuploidy simultaneously alters the copy number of many genes, aneuploidy affects more of the cancer genome than any other genetic alteration \cite{Ben-David2020-xy}. DNA copy number is highly correlated to RNA expression and protein production \cite{Pavelka2010-oh, Torres2007-ln}, providing a mechanism for aneuploidy to alter cellular phenotype. Therefore aneuploidy is thought to provide a substrate for tumour evolution \cite{Bakhoum2018-ai, Chen2012-kk, Shih2023-xy}. This notion is supported by the observation that chromosomes rich in oncogenes are frequently amplified whilst those rich in tumour suppressor genes are often deleted \cite{Davoli2013-kp}.  

The factors which explain aneuploidy patterns in cancers are not limited to the density of driver or suppressor genes on a particular chromosome. Aneuploidy is usually detrimental to cell fitness \cite{Torres2008-te}, in the first instance due to proteins such as P53 which cause apoptosis or cell cycle arrest in response to chromosome mis-segregations \cite{Tang2013-ki}. According to the \textit{gene doseage} hypothesis, aneuploidy can also reduce fitness by upsetting the balance of protein levels within cells: leading to negative effects such as impaired formation of stoichiometry dependent protein complexes, or protein aggregates that overwhelm protein quality-control mechanisms \cite{Torres2008-te,Tang2013-ki}. Thus aneuploidy patterns are context dependent \cite{Ben-David2020-xy}. Environmental context plays a role in sculpting karyotype, since the specific pressures of an environment will determine whether the fitness advantages of a particular CNA outweigh the costs. Evidence for the role of environment in determining karyotype includes the selective advantage of particular karyotypes under stressful conditions in yeast \cite{Pavelka2010-oh} and distinct patterns of aneuploidies between cancer types \cite{ICGCTCGA_Pan-Cancer_Analysis_of_Whole_Genomes_Consortium2020-hw, Nguyen2022-dg, Drews2022-sk}. Genomic context also plays a role in sculpting karyotype because a given CNA may only be favorable if other mutations or CNAs are already present within the cell. Evidence for the importance of genomic context includes observations that CNAs which are not independently significant predict survival when co-occurring \cite{Shukla2020-qi}, and defined temporal ordering of CNAs observed in a patient derived xenograft model \cite{Karlsson2022-rq, Baslan2022-ah}.

Aneuploidy remains difficult to study, for reasons which include the difficulty of experimentally inducing aneuploidy and the difficulty of distinguishing the effects of aneuploidy from those of chromosomal instability, the process which causes aneuploidy \cite{Ben-David2020-xy}. \textit{In silico} models will be an important tool to further our knowledge of aneuploidy. Gusev and colleagues developed the first model describing whole-chromosome mis-segregations \cite{Gusev2000-bj, Gusev2001-qc}. This model laid the mathematical foundations for describing segregation errors and explained patterns of aneuploidy in experimental data as a consequence of variable chromosome mis-segregation rate. A limitation of this model was that a fitness landscape defining the effect of aneuploidy on cell fitness was not considered, beyond a constraint that cells losing all copies  of any chromosome were not considered viable. This limitation was later addressed by others who assumed that the fitness effect of changing the copy number of a particular chromosome was dependent on the number of oncogenes or tumor suppressor genes expressed on that chromosome \cite{Elizalde2018-de,Laughney2015-qk}. In these models cell fitness could be increased by gaining additional copies of chromosomes with many oncogenes, or losing copies of chromosomes with many tumor suppressor genes. These models predicted an optimal mis-segregation rate (in the sense of minimising total cell death) that matched experimental observations, and resulted in a near-triploid karyotype that is frequently observed in tumour cells. All these previous models of aneuploidy share the limitation that they ignore the context dependency of the mapping between karyotype and fitness. This is perhaps unsurprising, since the vast number of possible karyotypes is challenging enough to map even without the additional variability introduced by context. However, the burgeoning quantity of single cell copy number data \cite{Salehi2021-kw,Karlsson2023-nu} now permits tracking of subclonal evolution at unprecedented resolution, giving the potential to refine our understanding of the relationships between karyotype and cellular fitness. 


Here, we present a procedure to directly estimate fitness landscapes from single-cell copy number data. We first develop an agent based model (ABM) of karyotype evolution on artificially generated fitness landscapes. We use output data from the ABM to demonstrate the predictive ability of our inference methodology. We go on to validate the predictive power of our method on previously published data from several P53 deficient cell lines which exhibited substantial subclonal evolution across multiple passages in vivo or in vitro \cite{Salehi2021-kw}. Finally, we analysed our fitted landscapes to study the context dependence of karyotype fitness, and how  missegregation rate can interact with fitness to determine clonal dominance. 

\section*{Discussion}


We and others have previously modelled karyotypic evolution via the process of missegregation \cite{Gusev2000-bj, Gusev2001-qc,Elizalde2018-de,Laughney2015-qk,Kimmel2023-th}. Across these modelling efforts, various assumptions have been made regarding fitness associated with specific karyotypes: either that all karyotypes are equally fit \cite{Gusev2001-qc}, that fitness is associated with density of driver or suppressor genes on each chromosome \cite{Elizalde2018-de,Laughney2015-qk}, or that fitness is negatively correlated with deviation from a euploid state \cite{Kimmel2023-th}. These attempts to model karyotypic fitness landscapes all ignore context, i.e. that the fitness landscape is sculpted by multiple factors including genetic background, tumour microenvironment, immune interactions and more\cite{Ben-David2020-xy}. This context dependency along with the large number of possible karyotypes makes reconstruction of fitness landscapes a challenging task. The mathematical model presented here offers the flexibility necessary to begin reconstructing adaptive fitness landscapes. Based on the dynamics of just a few subclones, our method is able to extrapolate the fitness of thousands of karyotypes. 

One application of our methodology is the study of fitness landscape topology. To that end we adapted  metrics from geostatistics to study the fitness landscape of a P53 deficient 184-hTERT diploid breast epithelial cell line\cite{Salehi2021-kw}. Over several months of passaging these cells developed substantial aneuploidy and underwent WGD, allowing us to compare the topology of the fitness landscape between the diploid and tetraploid states. It is thought that whole genome doubling promotes aneuploidy tolerance   \cite{Kuznetsova2015-fe,Prasad2022-bc}, a hypothesis which if true predicts a smoother fitness landscape around tetraploid landscapes compared to diploid counterparts. Indeed our topological analysis revealed a significantly larger Moran's i in the tetraploid landscape, in line with the notion that WGD ameliorates the fitness impacts associated SCNAs.

One limitation of our study is the lack of experimental data upon which our model predictions can be validated. In principle our model should be able to predict karyotypic evolution among aneuploid cell populations, however in practise this was prohibited by the paucity of data available to us. Nevertheless we do present a cross validation procedure that serves as a useful heuristic to evaluate the success of our method. Using this cross validation procedure we were able to demonstrate the ability of our method to predict the fitness of unobserved karyotypes across several cell lines with $R^2$ values in the range 0.4-0.6. Future work will incorporate data from studies with identical biological replicates \cite{Karlsson2023-nu}, which will offer opportunities for improved validation of model predictions. 

A second limitation of our model is the assumption that missegregation rate is homogeneous throughout a given fitness landscape. In particular, high ploidy cells may exhibit greater missegregation rates than low ploidy cells which could influence our conclusions surrounding the topology of the fitness landscapes near diploid and tetraploid karyotypes. Previously we have used interferon gamma as a measure of missegregation rate\cite{Kimmel2023-th}. Incorporating such a metric in future work could help us quantify the extent to which a permissive fitness landscape, rather than propensity to missegregate, explains observed associations between WGD and aneuploidy.

Future applications of this model will include more detailed studies of the fitness costs and benefits of high ploidy. In addition to expanding our characterisation of the relation between WGD and aneuploidy tolerance, our model will also help quantify the energetic requirements of high ploidy cells. Ultimately our model will be a powerful tool for studying karyotype evolution, revealing how selection acts upon coexisting karyotypes in various environments.


 

\bibliography{paperpile.bib}
\bibliographystyle{unsrt}
