---
title: "landscape characterization"
author: "Richard J Beck"
date: "`r Sys.Date()`"
output:
  md_document:
    variant: markdown_github
---
**Characterization of fitness landscapes**
```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir="~/projects/008_birthrateLandscape/ALFA-k")
```

```{r,echo=F,message=F,warning=F}
#source("utils/analysis_functions_v2.R")
source("utils/landscape_functions.R")
source("utils/ALFA-K.R")
library(ggplot2)
library(fields)

```

**1.** Is there evidence for (diminishing returns) epistasis?

Epistasis is defined as a "deviation from additivity in the effect of alleles at different loci with respect to their contribution to a quantitative phenotype"(Fisher, 1918 - see *Epistasis: what it means, what it doesn't mean, and statistical methods to detect it in humans. Heather J. Cordell.*). The logical extension of this definition to karyotype space implies that a chromosome can exhibit epistasis *with itself*. 


```{r,echo=F}

cn <- 1:8
alpha=0.1
df <- data.frame(cn,no.epistasis=alpha*cn,positive.epistasis=alpha/5*cn^2)
df <- reshape2::melt(df,id.vars="cn")
p <- ggplot(df,aes(x=cn,y=value,color=variable))+
  geom_line()+
  scale_x_continuous("copy number")+
  scale_y_continuous("fitness")+
  scale_color_discrete("")
p

```
In karyotype space, a mutation can be considered as a vector from one karyotype to another. If there is no epistasis, then any translation of that vector inside karyotype space should always be associated with the same delta fitness. It should be obvious then that for karyotype space the only fitness landscape that exhibits no epistasis is one in which fitness increases or decreases linearly as copy number of any chromosome changes. 
```{r,echo=F}
x <- expand.grid(x1=1:8,x2=1:8)
a <- 0.02
b <- 0.01

x$fitness <- x$x1*a+x$x2*b

p <- ggplot(x,aes(x=x1,y=x2,fill=fitness))+
  geom_raster()+
  scale_fill_viridis_c()+
  scale_x_continuous("chromosome 1")+
  scale_y_continuous("chromosome 2")
p

```

**Epistasis for hTERTa and hTERTb.** 
There are only 44 possible mutations in karyotype space (Chr1 gain, Chr1 loss, Chr2 gain...etc). Our approach will be ask what delta fitness is associated with each of the 44 mutations using each of the *frequent clones* as a starting point. We will then see if the distribution of delta fitnesses is dependent on starting fitness. 

```{r, echo=F}

idvec <- function(i){
  chr <- ceiling(i/2)
  sign <- c("+","-")[1+(1-i%%2)]
  chr <- stringr::str_pad(chr,2)
  paste0("chr.",chr,sign)
}


find_epistasis <- function(cellLine){
  x <- readRDS(paste0("example_data/",cellLine,"_fit.Rds"))
  k <- x$xo
  k <- rownames(k)[k$id=="fq"]
  kvec <- lapply(k,s2v)
  y <- lapply(kvec, function(ki) rbind(ki,t(apply(m,1,function(mi) mi+ki))))

  df <- do.call(rbind,lapply(1:length(y), function(i){
    f <- c(predict(x$fit,y[[i]]))
    f0 <- f[1]
    f <- f[-1]
    nullisomy <- apply(y[[i]],1,function(yij) sum(yij==0)>0)[-1]
    data.frame(k=k[i],fk=f0,df=f-f0,nullisomy=nullisomy,id=1:length(f))
  }))
  df$mut_id <- idvec(df$id)
  df$cellLine <- cellLine
  return(df)
}

m <- do.call(rbind,lapply(1:22, function(i){
  mi <- rep(0,22)
  mi[i] <- 1
  mi <- rbind(mi,-mi)
  return(mi)
}))


df1 <- find_epistasis("hTERTa")
df2 <- find_epistasis("hTERTb")


p1 <- ggplot(df2[!df2$nullisomy&df2$id<11,],aes(x=fk,y=df))+
  facet_wrap(~mut_id,ncol=2)+
  geom_point()+
  scale_x_continuous("fitness")+
  scale_y_continuous("delta fitness")+
  ggtitle("epistasis for selected mutations hTERTb")
p1

p2 <- ggplot(df1,aes(x=fk,y=df))+
  geom_point()+
  scale_x_continuous("fitness")+
  scale_y_continuous("delta fitness")+
  ggtitle(paste("hTERTa. correlation:",round(cor(df1$df,df1$fk),digits=3)))
p2

p3 <- ggplot(df2,aes(x=fk,y=df))+
  geom_point()+
  scale_x_continuous("fitness")+
  scale_y_continuous("delta fitness")+
  ggtitle(paste("hTERTb. correlation:",round(cor(df2$df,df2$fk),digits=3)))
p3


```
The results show some diminishing returns epistasis for hTERTb and little to no evidence of epistasis for hTERTa. An interpretation of this could be that hTERTb is nearing or has reached a peak in the fitness landscape whereas hTERTa is still climbing (??).

**2.** Characterizing landscape topology using Morans' I and "roughness" metric.

Roughness metric as defines as the mean absolute difference in fitness between a karyotype and all it's distance 1 neighbours:

```{r,echo=F}

x <- readRDS("example_data/hTERTcomb_fit.Rds")
df <- roughness_meas(x)

df$ploidyround <- round(df$ploidy)

print("mean roughness by ploidy (rounded to nearest integer)")
aggregate(list(roughness=df$roughness),by=list(ploidy=df$ploidyround),mean)

p <- ggplot(df,aes(x=ploidy,y=roughness))+
  geom_point()
p
```

Moran metric characterization:
```{r,echo=FALSE}

moran_freq <- function(karyotypes,fit,m=2){
  
  xmat <- do.call(rbind,lapply(karyotypes, function(i){
    as.numeric(unlist(strsplit(i,split="[.]")))
  }))
  fitness <- predict(fit,xmat)
  ploidy <- apply(xmat,1,mean)
  xbar <- mean(fitness)
  N <- length(fitness)
  d <- as.matrix(dist(xmat))
  w <- (1/d^m)#d*(d==1)
  diag(w) <- 0
  w <- apply(w,1,function(wi) wi/sum(wi))
  zi <- fitness-xbar
  
  m2 <- sum(zi^2)/N
  
  Ii <- sapply(1:length(fitness), function(i){
    zi[i]/m2*sum(w[i,]*zi)
  })  
  data.frame(Ii,ploidy)
}

x <- readRDS("example_data/hTERTcomb_fit.Rds")

karyotypes <- unique(unlist(lapply(x$xo,rownames)))
fit <- x$fit
df <- moran_freq(karyotypes,fit)
p <- ggplot(df,aes(x=ploidy,y=Ii))+
  geom_point()+
  scale_x_continuous("ploidy")+
  scale_y_continuous("Morans' I")
p
```
